File: leetcode/1.两数之和.java
import java.util.ArrayList;

/*
 * @lc app=leetcode.cn id=1 lang=java
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {
        boolean flag = false;
        int[] result = new int[2];
        for (int i = 0; i < nums.length; i++) {
            int sub = target - nums[i];
            result[0] = i;
            for (int j = i + 1; j < nums.length; j++) {
                if (nums[j] == sub) {
                    flag = true;
                    result[1] = j;
                    break;
                }
            }
            if (flag) {
                break;
            }
        }
        return result;
    }
}
// @lc code=end

File: 1.两数之和.go
/*
 * @lc app=leetcode.cn id=1 lang=golang
 *
 * [1] 两数之和
 */

// @lc code=start
func twoSum(nums []int, target int) []int {
	record := make(map[int]int)
	res := make([]int, 2)
	for i, v := range nums {
		complement := target - v
		val, ok := record[complement]
		if ok {
			res[0] = val
			res[1] = i
			break
		} else {
			record[v] = i
		}
	}
	return res
}
// @lc code=end

